### PR TITLE
Allow ignoring RBDMirror CR

### PR DESCRIPTION
**Pull request by umangachapagain**
Operator already allows CephRBDMirror metadata to be updated and does not override it on each reconcile.
If this is not enough for some reason, we can now ask the operator to ignore all changes to CephRBDMirror CR.

### CODE REVIEW

File: controllers/storagecluster/cephrbdmirrors.go [hunk 1]
```diff
@@ -82,6 +82,11 @@ func (r *StorageClusterReconciler) newCephRbdMirrorInstances(initData *ocsv1.Sto
 
 // ensureCreated ensures that cephRbdMirror resources exist in the desired state.
 func (obj *ocsCephRbdMirrors) ensureCreated(r *StorageClusterReconciler, instance *ocsv1.StorageCluster) (reconcile.Result, error) {
+	reconcileStrategy := ReconcileStrategy(instance.Spec.ManagedResources.CephRBDMirror.ReconcileStrategy)
+	if reconcileStrategy == ReconcileStrategyIgnore {
+		return reconcile.Result{}, nil
```
Comment (vbnrh) [inline controllers/storagecluster/cephrbdmirrors.go:87]: This looks like it only stops ocs from reconciling it ? Don't we need to label it as 'do_not_reconcile' for rook to ignore it as well ?

File: api/v1/storagecluster_types.go [hunk 2]
```diff
@@ -217,6 +218,11 @@ type ManageCephToolbox struct {
 	ReconcileStrategy string `json:"reconcileStrategy,omitempty"`
 }
 
+// ManageCephRBDMirror defines how to reconcile Ceph toolbox
```
Comment (iamniting) [inline api/v1/storagecluster_types.go:221]:
```suggestion
// ManageCephRBDMirror defines how to reconcile CephRBDMirror
```

**Review by iamniting — CHANGES_REQUESTED** [review]

**Review by iamniting — APPROVED** [review]

File: api/v1/storagecluster_types.go [hunk 3]
```diff
@@ -217,6 +218,11 @@ type ManageCephToolbox struct {
 	ReconcileStrategy string `json:"reconcileStrategy,omitempty"`
 }
 
+// ManageCephRBDMirror defines how to reconcile Ceph RBDMirror
+type ManageCephRBDMirror struct {
+	ReconcileStrategy string `json:"reconcileStrategy,omitempty"`
```
Comment (vbnrh) [inline api/v1/storagecluster_types.go:223]: I think it would be safer if you add few enum values for type safety in the CR

**Review by vbnrh — CHANGES_REQUESTED** [review]

File: api/v1/storagecluster_types.go (hunk 3 above)
Comment (vbnrh) [reply api/v1/storagecluster_types.go:223]: https://book.kubebuilder.io/reference/markers/crd-validation.html

**Pull request merged**